os/arch/arm/src/s5j/s5j_pm_idle.c

```diff
@@ -0,0 +1,122 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ *
+ *  arch/arm/src/s5j/s5j_pm_idle_init_data.c
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <errno.h>
+
+#include <tinyara/config.h>
+#include <tinyara/arch.h>
+#include <tinyara/board.h>
+#include <tinyara/irq.h>
+#include <tinyara/pm/pm.h>
+
+#include "s5j_clock.h"
+#include "s5j_pm.h"
+#include "up_arch.h"
+#include "up_internal.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: s5j_idle_pmnormal
+ *
+ * Description:
+ *   Enter NORMAL mode.
+ *
+ ****************************************************************************/
+static int s5j_idle_pmnormal(void)
+{
+	s5j_clk_pll_select_mux(true);
+
+	return errno;
+}
+
+/****************************************************************************
+ * Name: s5j_idle_pmstop
+ *
+ * Description:
+ *   Enter STOP mode.
+ *
+ ****************************************************************************/
+static int s5j_idle_pmstandby(void)
+{
+	s5j_clk_pll_select_mux(false);
+
+	return errno;
+}
+
+/****************************************************************************
+ * Name: s5j_idle_pmstandby
+ *
+ * Description:
+ *   Enter STANDBY mode.
+ *
+ * Input Parameters:
+ *   None
+ *
+ ****************************************************************************/
+static int s5j_idle_pmstop(void)
+{
+	/* FIXME: implement me */
+	asm("wfi");
+
+	/* won't get here */
+
+	return errno;
+}
+
+/****************************************************************************
+ * Public Function Prototypes
+ ****************************************************************************/
+/* Switch to the new sleep state suggested by the PM framework */
+int up_pmidle_switch_state(int state_ID)
+{
+	int ret = OK;
+
+	switch (state_ID) {
+		case S5J_NORMAL:
+			ret = s5j_idle_pmnormal();
+			break;
+
+		case S5J_STOP:
```

Comment:
With respect to existing there is complete swap of STANDBY(mux disable) state & STOP(wfi) state, May I know the reason ?
existing : PM_NORMAL(PLL mux enable) -> IDLE -> STANDBY(PLL mux disable) -> STOP("wfi" state)
New : PM_NORMAL(PLL mux enable) -> STOP("wfi" state) -> STANDBY(PLL mux disable)

This will impact on power savings.
what is your opinion ?

Reply:
@shadaksharikp here only names are swapped, the implementation of function is same
@pradeepns85 had suggested to keep names simplified where stop means last state & standby an intermediate one